### PR TITLE
Update icq-relayer to ibc v8

### PR DESCRIPTION
**Pull request by faddat**
## 1. Summary
Closes:

#1065

## 2.Type of change

Massive update of various dependencies for the relayer

## 3. Implementation details

This PR depends on #1060. It is an attempt to make the icq-relayer functional (lets say by testing against mainnet) with ibc v8 and sdk v0.50.x and cometbft v0.38.3

## 4. How to test/use

@baabeetaa should test this against `main`

## 6. Limitations (optional)

dunno if it will work right...

## Summary by CodeRabbit

- **Refactor**
    - Updated import paths across several files to reflect a version update and improve organization.
    - Modified module imports and initialization to align with dependency updates.
- **Chores**
    - Reorganized import statements for better code clarity without altering functionality.

### CODE REVIEW

File: icq-relayer/pkg/runner/run.go [hunk 1]
```diff
@@ -804,7 +805,7 @@ func unique(msgSlice []sdk.Msg, logger log.Logger) []sdk.Msg {
 }
 
 func Close() error {
-	query := tmquery.MustParse(fmt.Sprintf("message.module='%s'", "interchainquery"))
+	query := cmtquery.MustParse(fmt.Sprintf("message.module='%s'", "interchainquery"))
```
[the remark this answers is not in the document]
Comment (faddat) [reply icq-relayer/pkg/runner/run.go:808]: it is correct.

[the remark this answers is not in the document]
Comment (faddat) [reply icq-relayer/pkg/runner/run.go:808]: just did not update the code to reflect the use of the new cmtquery package name.  Tendermint was renamed comet.  I'll do that now though I'd be kind of astounded if you could

**Comment by faddat** [comment]
@coderabbitai what seems to be the issue with the icq relayer build?

**Comment by faddat** [comment]
@tropicaldog this looks good!

However, do you happen to know why `quicksilverd` is no longer building with these changes in place?

**Comment by tropicaldog** [comment]
@faddat all good sir, monorepo is tricky to manage to be honest

**Comment by faddat** [comment]
@tropicaldog before I approve this, could you change the module path from v7 to v8 for the relayer?  that way we can make it match the upstream ibc version instead of just matching quicksilver's version :)

**Comment by faddat** [comment]
@joe-bowman @chalabi2 -- can you guys check this out?

**Comment by faddat** [comment]
@chalabi2 can you confirm that this works for you?

After that, we can merge it.

**Comment by chalabi2** [comment]
> @chalabi2 can you confirm that this works for you?
> 
> After that, we can merge it.

Can confirm working on my end.

**Comment by joe-bowman** [comment]
This pull request has been deployed to Vercel.

<table>
<tr>
<td><strong>Latest commit:</strong></td>
<td><code>2bda0e7</code></td>
</tr>
<tr>
<td><strong>✅ Preview:</strong></td>
<td><a href='https://pr-1067.app-dev.quicksilver.zone'>https://pr-1067.app-dev.quicksilver.zone</a></td>
</tr>
<tr>
<td><strong>🔍 Inspect:</strong></td>
<td><a href='https://vercel.com/quicksilver/quicksilver/3atQhXRW4pnPaxZQH2Dk5CgMpX9E'>https://vercel.com/quicksilver/quicksilver/3atQhXRW4pnPaxZQH2Dk5CgMpX9E</a></td>
</tr>
</table>

[View Workflow Logs](https://github.com/quicksilver-zone/quicksilver/actions/runs/8463947301)

**Comment by joe-bowman** [comment]
This pull request has been deployed to Vercel.

<table>
<tr>
<td><strong>Latest commit:</strong></td>
<td><code>f0e9377</code></td>
</tr>
<tr>
<td><strong>✅ Preview:</strong></td>
<td><a href='https://pr-1067.app-dev.quicksilver.zone'>https://pr-1067.app-dev.quicksilver.zone</a></td>
</tr>
<tr>
<td><strong>🔍 Inspect:</strong></td>
<td><a href='https://vercel.com/quicksilver/quicksilver/7JA9mhcLCZg5NidwyWbiv3QcFcw8'>https://vercel.com/quicksilver/quicksilver/7JA9mhcLCZg5NidwyWbiv3QcFcw8</a></td>
</tr>
</table>

[View Workflow Logs](https://github.com/quicksilver-zone/quicksilver/actions/runs/8642710434)

**Comment by faddat** [comment]
@joe-bowman good to merge this?  @chalabi2  has tried it and it's happy++.

**Comment by joe-bowman** [comment]
This pull request has been deployed to Vercel.

<table>
<tr>
<td><strong>Latest commit:</strong></td>
<td><code>a68e3fb</code></td>
</tr>
<tr>
<td><strong>✅ Preview:</strong></td>
<td><a href='https://pr-1067.app-dev.quicksilver.zone'>https://pr-1067.app-dev.quicksilver.zone</a></td>
</tr>
<tr>
<td><strong>🔍 Inspect:</strong></td>
<td><a href='https://vercel.com/quicksilver/quicksilver/HWP9BcsdrZxbQUH2um1Z1Nyw9qCE'>https://vercel.com/quicksilver/quicksilver/HWP9BcsdrZxbQUH2um1Z1Nyw9qCE</a></td>
</tr>
</table>

[View Workflow Logs](https://github.com/quicksilver-zone/quicksilver/actions/runs/8643438589)

**Comment by joe-bowman** [comment]
This pull request has been deployed to Vercel.

<table>
<tr>
<td><strong>Latest commit:</strong></td>
<td><code>85cd280</code></td>
</tr>
<tr>
<td><strong>✅ Preview:</strong></td>
<td><a href='https://pr-1067.app-dev.quicksilver.zone'>https://pr-1067.app-dev.quicksilver.zone</a></td>
</tr>
<tr>
<td><strong>🔍 Inspect:</strong></td>
<td><a href='https://vercel.com/quicksilver/quicksilver/Gn36xB6WPFr6xmEe5xRc6wE4kYUY'>https://vercel.com/quicksilver/quicksilver/Gn36xB6WPFr6xmEe5xRc6wE4kYUY</a></td>
</tr>
</table>

[View Workflow Logs](https://github.com/quicksilver-zone/quicksilver/actions/runs/8653709108)

**Comment by joe-bowman** [comment]
This fails for me against dydx-mainnet-1:

```
ts=2024-04-15T17:50:58.923333408Z caller=run.go:641 chain=quicksilver-2 src_chain=dydx-mainnet-1 worker=historic msg="Error: Could not get header error: Could not get state from chain: \"no concrete type registered for type URL /ibc.lightclients.tendermint.v1.ClientState against interface *exported.ClientState\" "
```

**Comment by joe-bowman** [comment]
This pull request has been deployed to Vercel.

<table>
<tr>
<td><strong>Latest commit:</strong></td>
<td><code>b25dfab</code></td>
</tr>
<tr>
<td><strong>✅ Preview:</strong></td>
<td><a href='https://pr-1067.app-dev.quicksilver.zone'>https://pr-1067.app-dev.quicksilver.zone</a></td>
</tr>
<tr>
<td><strong>🔍 Inspect:</strong></td>
<td><a href='https://vercel.com/quicksilver/quicksilver/BdiYueprUCx7LeKA3Wtkc4KmhkFn'>https://vercel.com/quicksilver/quicksilver/BdiYueprUCx7LeKA3Wtkc4KmhkFn</a></td>
</tr>
</table>

[View Workflow Logs](https://github.com/quicksilver-zone/quicksilver/actions/runs/8706918543)

**Comment by joe-bowman** [comment]
This pull request has been deployed to Vercel.

<table>
<tr>
<td><strong>Latest commit:</strong></td>
<td><code>4c9e69d</code></td>
</tr>
<tr>
<td><strong>✅ Preview:</strong></td>
<td><a href='https://pr-1067.app-dev.quicksilver.zone'>https://pr-1067.app-dev.quicksilver.zone</a></td>
</tr>
<tr>
<td><strong>🔍 Inspect:</strong></td>
<td><a href='https://vercel.com/quicksilver/quicksilver/7XFriLYLTkVDSQmXgtgg1iDNkQwi'>https://vercel.com/quicksilver/quicksilver/7XFriLYLTkVDSQmXgtgg1iDNkQwi</a></td>
</tr>
</table>

[View Workflow Logs](https://github.com/quicksilver-zone/quicksilver/actions/runs/8707193065)

**Comment by joe-bowman** [comment]
This pull request has been deployed to Vercel.

<table>
<tr>
<td><strong>Latest commit:</strong></td>
<td><code>793c25a</code></td>
</tr>
<tr>
<td><strong>✅ Preview:</strong></td>
<td><a href='https://pr-1067.app-dev.quicksilver.zone'>https://pr-1067.app-dev.quicksilver.zone</a></td>
</tr>
<tr>
<td><strong>🔍 Inspect:</strong></td>
<td><a href='https://vercel.com/quicksilver/quicksilver/BACA8pWxbfVrsNUaABvEp64UUVEa'>https://vercel.com/quicksilver/quicksilver/BACA8pWxbfVrsNUaABvEp64UUVEa</a></td>
</tr>
</table>

[View Workflow Logs](https://github.com/quicksilver-zone/quicksilver/actions/runs/8707301625)

**Comment by joe-bowman** [comment]
This pull request has been deployed to Vercel.

<table>
<tr>
<td><strong>Latest commit:</strong></td>
<td><code>5b8fd34</code></td>
</tr>
<tr>
<td><strong>✅ Preview:</strong></td>
<td><a href='https://pr-1067.app-dev.quicksilver.zone'>https://pr-1067.app-dev.quicksilver.zone</a></td>
</tr>
<tr>
<td><strong>🔍 Inspect:</strong></td>
<td><a href='https://vercel.com/quicksilver/quicksilver/6iscLKcqr3Ms5nLEfbH36tVCGUY7'>https://vercel.com/quicksilver/quicksilver/6iscLKcqr3Ms5nLEfbH36tVCGUY7</a></td>
</tr>
</table>

[View Workflow Logs](https://github.com/quicksilver-zone/quicksilver/actions/runs/8707421990)

**Comment by joe-bowman** [comment]
This pull request has been deployed to Vercel.

<table>
<tr>
<td><strong>Latest commit:</strong></td>
<td><code>4ef55d8</code></td>
</tr>
<tr>
<td><strong>✅ Preview:</strong></td>
<td><a href='https://pr-1067.app-dev.quicksilver.zone'>https://pr-1067.app-dev.quicksilver.zone</a></td>
</tr>
<tr>
<td><strong>🔍 Inspect:</strong></td>
<td><a href='https://vercel.com/quicksilver/quicksilver/9tpCXDXdz72up5mpoxSEee1GsBwr'>https://vercel.com/quicksilver/quicksilver/9tpCXDXdz72up5mpoxSEee1GsBwr</a></td>
</tr>
</table>

[View Workflow Logs](https://github.com/quicksilver-zone/quicksilver/actions/runs/8707441337)

**Comment by joe-bowman** [comment]
This pull request has been deployed to Vercel.

<table>
<tr>
<td><strong>Latest commit:</strong></td>
<td><code>ac27eaa</code></td>
</tr>
<tr>
<td><strong>✅ Preview:</strong></td>
<td><a href='https://pr-1067.app-dev.quicksilver.zone'>https://pr-1067.app-dev.quicksilver.zone</a></td>
</tr>
<tr>
<td><strong>🔍 Inspect:</strong></td>
<td><a href='https://vercel.com/quicksilver/quicksilver/HWnJgdLAJn4fBoXJ2hiGhr8wjKdZ'>https://vercel.com/quicksilver/quicksilver/HWnJgdLAJn4fBoXJ2hiGhr8wjKdZ</a></td>
</tr>
</table>

[View Workflow Logs](https://github.com/quicksilver-zone/quicksilver/actions/runs/8708706771)

**Comment by joe-bowman** [comment]
This pull request has been deployed to Vercel.

<table>
<tr>
<td><strong>Latest commit:</strong></td>
<td><code>7f86cf7</code></td>
</tr>
<tr>
<td><strong>✅ Preview:</strong></td>
<td><a href='https://pr-1067.app-dev.quicksilver.zone'>https://pr-1067.app-dev.quicksilver.zone</a></td>
</tr>
<tr>
<td><strong>🔍 Inspect:</strong></td>
<td><a href='https://vercel.com/quicksilver/quicksilver/GUCf916DihoPyRnSaVeUfCmStH7N'>https://vercel.com/quicksilver/quicksilver/GUCf916DihoPyRnSaVeUfCmStH7N</a></td>
</tr>
</table>

[View Workflow Logs](https://github.com/quicksilver-zone/quicksilver/actions/runs/8708957625)

**Comment by joe-bowman** [comment]
This pull request has been deployed to Vercel.

<table>
<tr>
<td><strong>Latest commit:</strong></td>
<td><code>c471c0a</code></td>
</tr>
<tr>
<td><strong>✅ Preview:</strong></td>
<td><a href='https://pr-1067.app-dev.quicksilver.zone'>https://pr-1067.app-dev.quicksilver.zone</a></td>
</tr>
<tr>
<td><strong>🔍 Inspect:</strong></td>
<td><a href='https://vercel.com/quicksilver/quicksilver/2DAMY65ojSWcX3vigcykVabSfjha'>https://vercel.com/quicksilver/quicksilver/2DAMY65ojSWcX3vigcykVabSfjha</a></td>
</tr>
</table>

[View Workflow Logs](https://github.com/quicksilver-zone/quicksilver/actions/runs/8708973356)

**Pull request closed without merging**